src/base/bittorrent/session.cpp

```diff
@@ -845,7 +845,7 @@ void Session::setSubcategoriesEnabled(bool value)
     emit subcategoriesSupportChanged();
 }
 
-QSet<QString> Session::tags() const
+const QSet<QString> &Session::tags() const
```

Comment:
@thalieht 
Now I suspect the [problem](https://github.com/qbittorrent/qBittorrent/pull/9824#discussion_r232918799) returns.

Also ping @glassez, the same apply to your change at https://github.com/qbittorrent/qBittorrent/pull/9883/files#diff-b369e532eff50253387c9d203da4b928R453

Reply:
We shouldn't modify containers inside range based `for` loop! In such case we should either iterate over its copy or use another way.

Reply:
Now I suspect "Return const reference for a few shared containers" is not going the right way, it changes the whole semantics of the function... aren't we used to getting a new copy of container from Qt functions? Now we only get constant references, it is a big difference...

Reply:
>aren't we used to getting a new copy of container from Qt functions? Now we only get constant references, it is a big difference...

Damn. You're right.
So what are we supposed to do? There are two main options:
1. Apply `copyAsConst()` in a "paranoid way" to all rvalues.
2. Apply `copyAsConst()` only to the rvalues that produces shared copy of inner containers (if this follows from the meaning of such method, because we should not get into its implementation).

Reply:
Will delete this commit.